presto-main/src/test/java/io/prestosql/server/security/oauth2/TestOAuth2WebUiAuthenticationFilter.java

```diff
@@ -0,0 +1,407 @@
+            WebDriverWait wait = new WebDriverWait(driver, 5);
+            submitCredentials(driver, "foo@bar.com", "foobar", wait);
+            giveConsent(driver, wait);
+            wait.until(ExpectedConditions.urlMatches(format("https://host.testcontainers.internal:%d/ui/", HTTPS_PORT)));
+
+            assertion.assertWith(driver, wait);
+        }
+    }
+
+    private BrowserWebDriverContainer<?> createChromeContainer()
+    {
+        ChromeOptions options = new ChromeOptions();
+        options.setAcceptInsecureCerts(true);
+        BrowserWebDriverContainer<?> chromeContainer = new BrowserWebDriverContainer<>()
+                .withNetwork(testingHydraService.getNetwork())
+                .withCapabilities(options);
+        chromeContainer.start();
+        return chromeContainer;
+    }
+
+    private void submitCredentials(WebDriver driver, String email, String password, WebDriverWait wait)
+    {
+        By emailElementLocator = By.id("email");
+        System.out.println(driver.getCurrentUrl());
+        System.out.println(driver.getPageSource());
+        wait.until(elementToBeClickable(emailElementLocator));
+        WebElement usernameElement = driver.findElement(emailElementLocator);
+        usernameElement.sendKeys(email);
+        By passwordElementLocator = By.id("password");
+        wait.until(elementToBeClickable(passwordElementLocator));
+        WebElement passwordElement = driver.findElement(passwordElementLocator);
+        passwordElement.sendKeys(password + "\n");
+    }
+
+    private void giveConsent(WebDriver driver, WebDriverWait wait)
+    {
+        By openIdCheckboxLocator = By.id("openid");
+        wait.until(elementToBeClickable(openIdCheckboxLocator));
+        WebElement openIdCheckbox = driver.findElement(openIdCheckboxLocator);
+        openIdCheckbox.click();
+        By acceptButtonLocator = By.id("accept");
+        wait.until(elementToBeClickable(acceptButtonLocator));
+        WebElement acceptButton = driver.findElement(acceptButtonLocator);
+        acceptButton.click();
+    }
+
+    private void assertPrestoCookie(Cookie cookie)
+    {
+        assertThat(cookie.getName()).isEqualTo("Presto-OAuth2-Token");
+        assertThat(cookie.getDomain()).isEqualTo("host.testcontainers.internal");
+        assertThat(cookie.getPath()).isEqualTo("/ui/");
+        assertThat(cookie.isSecure()).isTrue();
+        assertThat(cookie.isHttpOnly()).isTrue();
+        assertThat(cookie.getValue()).isNotBlank();
+        Jws<Claims> jwt = Jwts.parser()
+                .setSigningKeyResolver(new JwkSigningKeyResolver(new JwkService(
+                        new JwtAuthenticatorConfig().setKeyFile("http://localhost:" + testingHydraService.getHydraPort() + "/.well-known/jwks.json"),
+                        new JettyHttpClient())))
+                .parseClaimsJws(cookie.getValue());
+        // TODO: check source of the difference
```

Comment:
Should this be addressed? Seems like it could be a bug

Reply:
The difference is only 1 second and I believe it's due to selenium's rounding of the expiration time.

Reply:
Can we change the test to make sure it's within a bound? Say within 5 seconds?